tests/mdns/main.cpp

```diff
@@ -98,7 +98,7 @@ void PublishSingleServiceWithCustomHost(void *aContext, Mdns::Publisher::State a
         Mdns::Publisher::TxtList txtList{
             {"nn", "cool"}, {"xp", xpanid, sizeof(xpanid)}, {"tv", "1.1.1"}, {"dd", extAddr, sizeof(extAddr)}};
 
-        error = sContext.mPublisher->PublishHost(hostName, hostAddr, sizeof(hostAddr));
+        error = sContext.mPublisher->PublishHost(hostName, {hostAddr, hostAddr + sizeof(hostAddr)});
```

Comment:
nit: make `hostAddr` a vector so you don't need to repeat this pattern in the test code.